### ISSUE TITLE
Permanent redirect (301) for old alias

### ISSUE THREAD

**Issue by GioviQ** [OrchardCMS/Orchard#6261]
For SEO purposes sometimes I have to change the urls of my pages. At the moment the old urls are lost, but to me they should be kept in alias table and used to generate permanent redirect (301) to current url, so Google and I are happy.
Now to achieve this I have to use this module http://orchardgalleryfe.cloudapp.net/List/Modules/Orchard.Module.Nublr.CustomAlias
It would be better to include this little feature in the native alias module.

**Comment by dcinzona** [comment]
That module has some bugs in it (multiple instances not getting the new alias, etc.).  That said, we currently use it as well.

**Comment by GioviQ** [comment]
Do you mean Custom Alias module has bugs? At the moment it seems to work well.

**Comment by dcinzona** [comment]
Yes

On Jan 8, 2016, at 3:49 AM, GioviQ <notifications@github.com<mailto:notifications@github.com>> wrote:


Do you mean Custom Alias module has bugs? At the moment it seems to work well.

-

**Comment by GioviQ** [comment]
I have the source code, do you know where is the issue?

**Comment by dcinzona** [comment]
Well, there are 2 issues.  One is incredibly minor (when saving a new alias, the redirect is a bit messed up and doesn't display a notification that the alias was saved.
The second one is the bigger issue, that I have not been able to resolve but have a work around:
When saving a new alias *in a multi-server environment* the alias works on one instance, but not on the other instances.  I've tried a bunch of different things, and from what I can tell, it's not a caching issue.  The only way to work around the issue, from what I can tell, is to connect to the other instance and edit the alias there, then save it again on that instance (or restart the shell).

**Comment by sfmskywalker** [comment]
> When saving a new alias in a multi-server environment the alias works on one instance, but not on the other instances

I think that's where the **Alias Updater** feature comes in. It's disabled by default, but should be enabled in multi-server environments.

**Comment by dcinzona** [comment]
Does CustomAlias use that feature?  I will try it to see if it solves the issue

**Comment by dcinzona** [comment]
Just checked and Alias Updater is was already enabled :\

**Comment by GioviQ** [comment]
When you save a new redirect, you do not have notification, but the form title change in Edit.

**Comment by dcinzona** [comment]
yeah, it's missing the notification event in the create method.  Simple fix.  The web farm issue is the bigger concern and just something you need to be aware of.

**Comment by GioviQ** [comment]
Ok, I only have one server...

**Comment by sebastienros** [comment]
The recommendation is to use a dedicated module for permanent redirections.
